### PR TITLE
fix: crash when value is `undefined`

### PR DESCRIPTION
**Pull request by parasharrajat**
### Details

`ComponentDidUpdate` was allowing the undefined value to be pass down to native TextInput value which was causing an issue. So I added a check if that value prop is `undefined` (~not provided), skip the update.

### Fixed Issues

$ https://github.com/Expensify/App/issues/7769
$ https://github.com/Expensify/App/issues/7774
$ https://github.com/Expensify/App/issues/7775
$ https://github.com/Expensify/App/issues/7869

### Tests
1. Open the Payments page and tried to set a payment method as default and the password popup should come up and no errors.

- [x] Verify that no errors appear in the JS console

### QA Steps
1. Go to the payments page. 
2. Click any existing payment method to set it to default. 
3. Click set default and no error should pop up.

- [x] Verify that no errors appear in the JS console

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web | Desktop

|||
|-------|-------|
|image|image|

#### Mobile Web

image

#### iOS

image

#### Android

image

### CODE REVIEW

**Comment by parasharrajat** [comment]
Updating screens shortly.....testing more issues.

**Comment by parasharrajat** [comment]
I will update the IOS screen tomorrow. But I am pretty sure it is working.

**Comment by Julesssss** [comment]
Sorry, @parasharrajat I tried to verify this for you but I've had a lot of problems trying to verify this, none of which are related to the PR 😩 

First I created some bank accounts with the generator script. Then I created an Expensify wallet. Next, I upgraded it to Gold. But I'm still not able to link a bank account to a wallet... All three of my accounts show that I'm [using an invalid bank account](https://github.com/Expensify/Auth/blob/fbe360030da875a10441ebc088213d93759c5f3c/auth/command/SetWalletLinkedAccount.cpp#L54) when setting them as the default, but the error doesn't explain why. 

I modified the Auth code in order to give me the exact reason why, but now I'm getting 666 after updating and need to reprovision as updateRepos w/ build didn't work either. My last attempt will be to restart my machine, which I'll do after sending this.

**Comment by parasharrajat** [comment]
I was able to skip the App freezing issue and Test it and it is not crashing the app. I will attach the screen shotly.

**Comment by Beamanator** [comment]
Sorry guys I'm pretty heads down in a design doc I didn't have time to test last weekend and may not have time today either :'( If @Julesssss is having many problems we can ping another engineer to help test

**Comment by sketchydroide** [comment]
I'll try testing this today

**Comment by nkuoch** [comment]
Just FYI, I've been using this fix in the PRs I am working on because I had the bug, and it works well. (https://github.com/Expensify/App/pull/7804 and https://github.com/Expensify/App/pull/7813) ... and would love to see it merged ASAP so I can merge it in my PRs :)

**Comment by parasharrajat** [comment]
Thanks, @nkuoch for your input.

**Comment by timszot** [comment]
Alright, so this looks to solve our [3 deploy blocker issues](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3ADeployBlockerCash). @nkuoch you said you've been testing this in relation to your own changes, which is great. Do you have time to review/merge this PR since the two assigned haven't had a chance to get to it so we can unblock the deploy?

**Review by nkuoch — APPROVED** [review]

File: src/components/TextInput/BaseTextInput.js [hunk 1]
```diff
@@ -59,7 +59,8 @@ class BaseTextInput extends Component {
 
     componentDidUpdate() {
         // Activate or deactivate the label when value is changed programmatically from outside
-        if (this.value === this.props.value) {
+        // Only update when value prop is provided
+        if (this.props.value === undefined || this.value === this.props.value) {
```
Comment (stitesExpensify) [inline src/components/TextInput/BaseTextInput.js:63]: NAB `_.isUndefined(this.props.value)`

Comment (parasharrajat) [reply src/components/TextInput/BaseTextInput.js:63]: Yeah, I will do that in https://github.com/Expensify/App/issues/6584 during refactoring.

**Pull request merged**